drivers/adc/adc_mcux_lpadc.c

```diff
@@ -496,7 +496,9 @@ static int mcux_lpadc_init(const struct device *dev)
 	adc_config.conversionAverageMode = config->calibration_average;
 #endif /* FSL_FEATURE_LPADC_HAS_CTRL_CAL_AVGS */
 
+#if (DT_INST_NODE_HAS_PROP(0, power_level))
```

Comment:
this is not the way to do this, what I meant is to add like a boolean property which would be set in the SOC level that says if the ip instance has this feature. and then you could add a BUILD_ASSERT in the init macro at the bottom of the file to check if the power level selection prop has been given despite not having the feature. And in the code here you should not use a preprocessor #if but use in C code `if (DT_ANY_INST_HAS_PROP_STATUS_OKAY(....)) {`

Reply:
Sorry, I accidentally clicked the "review again" button, and I will modify the code according to your comment later.

Reply:
What is the purpose of BUILD_ASSERT? For LPADC, much platforms have power level selection options, however rt1180 not. What criteria do I use in BUILD_ASSERT?

Reply:
> this is not the way to do this, what I meant is to add like a boolean property which would be set in the SOC level that says if the ip instance has this feature. and then you could add a BUILD_ASSERT in the init macro at the bottom of the file to check if the power level selection prop has been given despite not having the feature. And in the code here you should not use a preprocessor #if but use in C code `if (DT_ANY_INST_HAS_PROP_STATUS_OKAY(....)) {`

Can't use in C code because powerLevelMode element is used a preprocessor #if to isolation in lpadc_config_t structure. Otherwise, C compiler will report a error:   'lpadc_config_t' has no member named 'powerLevelMode'.
`#if defined(FSL_FEATURE_LPADC_HAS_CFG_PWRSEL) && (FSL_FEATURE_LPADC_HAS_CFG_PWRSEL == 1U)
    lpadc_power_level_mode_t powerLevelMode; /*!< Power Configuration Selection. */
#endif /* defined(FSL_FEATURE_LPADC_HAS_CFG_PWRSEL) && (FSL_FEATURE_LPADC_HAS_CFG_PWRSEL == 1U) */`

Reply:
i think we have a misunderstanding here, I'm confused what you're trying to do, the build assert should be checking a value from DT is what I meant

Reply:
> i think we have a misunderstanding here, I'm confused what you're trying to do, the build assert should be checking a value from DT is what I meant

Add BUILD_ASSERT  at the bottom of macro.